src/tool/xmeta/cpp/models/namespace_model.cpp

```diff
@@ -0,0 +1,51 @@
+#include "namespace_model.h"
+
+namespace xlang::xmeta
+{
+#pragma region namespace_body_model
+    bool namespace_body_model::member_id_exists(std::string_view const& member_id)
+    {
+        return classes.find(member_id) != classes.end() ||
+            structs.find(member_id) != structs.end() ||
+            interfaces.find(member_id) != interfaces.end();
+    }
+
+    std::string namespace_body_model::get_full_namespace_name()
+    {
+        return containing_namespace->get_full_namespace_name();
+    }
+#pragma endregion
+
+
+#pragma region namespace_model
```

Comment:
I think this pragma is VS specific and not appropriate for a portable library. It's also not very beneficial in such a small file.